nginx/files/redirect.conf

```diff
@@ -6,6 +6,6 @@ server {
   {%- if site.redirect is defined %}
   rewrite ^(.*)   {{ site.redirect.get('protocol', 'http') }}://{{ site.redirect.get('host', '$server_name') }}{{ site.redirect.get('path', '$1') }} permanent;
   {%- else %}
-  rewrite ^(.*)   https://$server_name$1 permanent;
+  rewrite ^(.*)   https://$http_host$1 permanent;
```

Comment:
$http_host can be undefined (HTTP/1.0) or defined with port, for example - the redirect from http://<...>:8080 to http**s**://<...>:8080 would not work.

The underscore is a catch all $server_name, maybe it should be avoided or we could use [$host](http://nginx.org/en/docs/http/ngx_http_core_module.html#var_host) variable to at least somewhat mitigate the issue.